execute-async-function.js

```diff
@@ -52,27 +52,46 @@
         return execArgs;
     }
 
-    /** Create a promise that resolves when chrome.runtime.onMessage fires with the id
+    /** Create a promise that resolves when a chrome event (such as chrome.runtime.onMessage) fires
+     * @param {Event} chromeEvent The event type.
      * @param {string} id ID for the message we're expecting.
      * Messages without the ID will not resolve this promise.
      * @returns {Promise} Promise that resolves when chrome.runtime.onMessage.addListener fires. */
-    function promisifyRuntimeMessage(id) {
-        // We don't have a reject because the finally in the script wrapper should ensure this always gets called.
+    function promisifyChromeEvent(chromeEvent, id) {

```

Comment:
There isn't that much code shared between the two `if` blocks. Would this be easier to maintain as `promisifyRuntimeMessage` and `promisifyTabUpdate`?